### PR TITLE
Add directory creator

### PR DESCRIPTION
**Pull request by y-myajima**
# Description

We have modularized the process of creating directories for outputting Access Token, Role Token, and Role Cert.

Related issue: https://github.com/AthenZ/k8s-athenz-sia/issues/49

## Assignees
- [x] `Assignees` is set

## Type of changes
- [x] Apply one or more `labels` of the following that fits:
  - `bug`: Bug fix
  - `dependencies`: Dependency upgrades
  - `documentation`: Documentation changes
  - `enhancement`: New Feature
  - `good first issue`: First contribution
  - `logging`: Log changes
  - `refactor`: Refactoring (no functional changes, no api changes)

### Flags
```
- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code
```
---

## Checklist
```
- [ ] Followed the guidelines in the CONTRIBUTING document
- [ ] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [ ] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation
```

## Checklist for maintainer
```
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge
```

### CODE REVIEW

File: pkg/util/directory-creator.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		log.Debugf("Creating new directory: %s", dir)
+		err = os.MkdirAll(dir, 0755)
+		if err != nil {
+			return fmt.Errorf("unable to create directory: %w", err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("unable to check directory: %w", err)
+	}
+	return nil
+}
```
Comment (mlajkim) [inline pkg/util/directory-creator.go:39]: Kind of nested with `if` and `else if` stuff..
the logic might be wrong so please double check:
```suggestion
func CreateDirectory(path string) error {
	dir := filepath.Dir(path)
	_, err := os.Stat(dir)
	if err == nil {
		return nil // directory already exists
	}

	if !os.IsNotExist(err) { // if it is not related to directory not found
		return fmt.Errorf("unable to check directory: %w", err)
	}

	log.Debugf("Creating new directory: %s", dir)
	err = os.MkdirAll(dir, 0755) // create directory
	if err != nil {
		return fmt.Errorf("unable to create directory: %w", err)
	}

	return nil // directory created successfully
}
```

**Review by mlajkim — COMMENTED** [review]
quick review

File: pkg/util/directory-creator.go (hunk 1 above)
Comment (y-myajima) [reply pkg/util/directory-creator.go]: fixed in: https://github.com/AthenZ/k8s-athenz-sia/pull/142/commits/a4f81626755426412559ddb9dfa4e42e53140c50

**Review by mlajkim — APPROVED** [review]
ok

**Comment by mlajkim** [comment]
This PR introduced a bug and has been addressed in: https://github.com/AthenZ/k8s-athenz-sia/pull/146

**Pull request merged**